gateway-server/src/main/java/org/apache/knox/gateway/services/DefaultGatewayServices.java

```diff
@@ -51,6 +50,33 @@
 
   private static GatewayMessages log = MessagesFactory.get( GatewayMessages.class );
 
+  private static final String[] SERVICE_START_ORDER = new String[]{
```

Comment:
Not sure if it is easy to reverse the array. Might be easier to do `Arrays.asList(...)`. Your comment about reversing the list for stop makes sense to me.

Side note - I've wanted to do this for a long time. Even on the init method. Init is harder since there are dependencies but still seems like it should be possible.

Reply:
Not saying we should change init with this PR, just commenting that I like the cleanup :)

Reply:
I looked at the init method too, but there is a bit more going on.  

As for the reverse, the stop order was different than the start order. So right now stop is not the same as start and would not be the exact reverse as start.  I am not sure what the order really should be, maybe we need to look at the services and declare an official order.  Ideally items will stop in the reverse order of how they were started.  We do not want to stop a service that is a dependent of some running service.